topdown/http.go

```diff
@@ -257,11 +257,7 @@ func useSocket(rawURL string, tlsConfig *tls.Config) (bool, string, *http.Transp
 	}
 	tr.TLSClientConfig = tlsConfig
 
-	if tlsConfig != nil {
-		rawURL = strings.Replace(rawURL, "unix:", "https:", 1)
-	} else {
-		rawURL = strings.Replace(rawURL, "unix:", "http:", 1)
-	}
+	rawURL = strings.Replace(rawURL, "unix:", "http:", 1)
```

Comment:
Would we ever need to specify https when using a local socket?  tlsConfig is always != nil.

Reply:
Is it? From looking at the code, it seems like `nil` could be passed here: 

https://github.com/open-policy-agent/opa/pull/3667/files#diff-4a137f284e9009d157ac4553842cb1c6e9260173fa29d3ed0d779f79e5b7a316R505-R510

But I might be missing something... 🤔

Reply:
In this section https://github.com/open-policy-agent/opa/blob/d5acefe8aedfd0caf25ef1903b288b9018729b3c/topdown/http.go#L448-L451 isTLS always seems to get set to "true".

Reply:
`tlsUseSystemCerts` defaults to true, but it *could be false* if configured so, via passing `tls_use_system_certs: false` to `http.send`.